src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -64,10 +66,15 @@ private[health] class HealthCheckActor(
       app.version,
       healthCheck
     )
-    val activeInstanceIds: Set[Instance.Id] = instanceTracker.specInstancesLaunchedSync(app.id).map(_.instanceId)(collection.breakOut)
-    // The Map built with filterKeys wraps the original map and contains a reference to activeInstanceIds.
-    // Therefore we materialize it into a new map.
-    healthByInstanceId = healthByInstanceId.filterKeys(activeInstanceIds).iterator.toMap
+    instanceTracker.instancesBySpec().map(
+      _.specInstances(app.id).filter(_.isLaunched).map(_.instanceId)(collection.breakOut)
+    ).onComplete { r =>
```

Comment:
How this should work with FSM? We will get events. If health event needs data from TaskTracker it is stashed. When actor receive message with TaskTracker data it process all stashed messages. Am I correct?